### DISCUSSION TITLE
Custom validator for vminterface.name stopped working on update to 3.3.8

### DISCUSSION THREAD

**Discussion by bluikko** [netbox-community/netbox#11031] [Q&A] [answered]
After updating to NetBox 3.3.8 from 3.2.x, now my custom validation logic that tries to validate the interface name with `instance.name` stopped working. It seems `instance.name` is an empty string every time?
Other validators work, other fields of vminterface can be validated. Anyone have idea why this would happen? Looking at release notes revealed nothing interesting and I see that the `VMInterface` model still has the `name` field.
Could it be a bug?

**Comment by kkthxbye-code** [comment, marked as answer]
The following works for me on v3.3.8:

```
from extras.validators import CustomValidator

class MyValidator(CustomValidator):
    def validate(self, instance):
        self.fail(f"Name is {instance.name}")

CUSTOM_VALIDATORS = {
    'virtualization.vminterface': (
        MyValidator(),
    )
}
```

When saving the device I get the interface name in the validation error message.

A simple non-code validator also works:

```
CUSTOM_VALIDATORS = {
    "virtualization.vminterface": [
        {
            "name": {
                "min_length": 5,
                "max_length": 10
            }
        }
    ]
}
```

You need to post your code if anyone is to replicate it.

**Reply by bluikko** [reply]
For me with this code in `MyValidator` the `instance.name` is also empty.

In configuration:
```python
CUSTOM_VALIDATORS = {
    'ipam.ipaddress': (
        'myvalidation.ValidateAddress',
    ),
    'virtualization.vminterface': (
        'myvalidation.ValidateInterface',
    )
}
```

And validator from `myvalidation.py`:
```python
class ValidateInterface(CustomValidator):

    def validate(self, instance):
        fail_message = []

        self.fail(f"Name is {instance.name}")
```

**Reply by kkthxbye-code** [reply]
I was missing that this only happens on create. This is indeed a bug, seemingly caused by https://github.com/netbox-community/netbox/pull/10312 which was included in v3.3.3.

Replication fields (think interface name expansion) was refactored in that PR and seems to have caused the name field to be blank at  the time of validation. Validation was also slight broken before v3.3.3 though according to my testing, the validation logic was triggered twice, the first time with a valid name and the second time with a blank name.

Please create an issue.

**Reply by bluikko** [reply]
Thanks.
